### ISSUE TITLE
Could not connect to the helper app

### ISSUE THREAD

**Issue by Wickywow** [crystalidea/macs-fan-control#226]
Macs Fan Control could not connect to the helper app to enable fan control as well as adjust the speeds. Make sure to enter the administrator password when asked.
<img width="448" alt="Screenshot 2019-11-02 at 08 55 46" src="https://user-images.githubusercontent.com/57282608/68067376-a6d89900-fd4e-11e9-92af-5f50a10a693e.png">

This is a pro version

**Comment by kleuter** [comment]
Did you enter the correct administrator password after launching the app?

**Comment by Wickywow** [comment]
> Did you enter the correct administrator password after launching the app?

HI Kleuter, yes I did

**Comment by kleuter** [comment]
Did the app ever work? I mean, when you installed it, purchase Pro version, you saw this warning all the time? Or did this warning appear after something?
Please post technical information from the Preferences dialog.

**Comment by Wickywow** [comment]
I only bought it now so I cant say if it never worked. First time I am installing and trying to use it.

Tech Info:

**Macs Fan Control 1.5.0 [Qt 5.13.1]
MacBookPro14,3
CPU: Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz [4 cores]
GPU: Intel HD Graphics 630, Radeon Pro 560
macOS 10.15.1

Fans:
Left side-min{2160}curr{2158}max{5927}mode{0}
Right side-min{2000}curr{2000}max{5489}mode{0}


TempSensors:
TA0V () - 30.3398, 30.3398
TB0T (Battery Max) - 30.5977, 30.5977
TB1T (Battery Sensor 1) - 29.0977, 29.0977
TB2T (Battery Sensor 2) - 30.5977, 30.5977
TBXT () - 30.5977, 30.5977
TC0E () - 53.8086, 53.8086
TC0F () - 55.2422, 55.2422
TC0P (CPU Proximity) - 45, 45
TC1C (CPU Core 1) - 51, 51
TC2C (CPU Core 2) - 49, 49
TC3C (CPU Core 3) - 49, 49
TC4C (CPU Core 4) - 49, 49
TCGC (GPU PECI) - 50, 50
TCMc () - 128, 128
TCSA () - 50, 50
TCTD () - 255.945, 255.945
TCXC (CPU PECI) - 51.1719, 51.1719
TG0D (GPU Diode) - 48.5, 48.5
TG0F () - 50.0664, 50.0664
TG0P (GPU Proximity) - 44, 44
TGDD () - 45, 45
TGVP () - 42.625, 42.625
TH0A () - 35.75, 35.75
TH0B () - 34.375, 34.375
TH0C () - 33.625, 33.625
TH0F () - 213.375, 213.375
TH0R () - 213.375, 213.375
TH0a () - 35.75, 35.75
TH0b () - 34.375, 34.375
TH0c () - 33.625, 33.625
TH0x () - 129, 129
TM0P (Memory Proximity) - 40.625, 40.625
TPCD (Platform Controller Hub Die) - 34, 34
TTLD () - 30.875, 30.875
TTRD () - 30.375, 30.375
TW0P (Airport Card) - 44.0625, 44.0625
TaLC () - 31.8125, 31.8125
TaRC () - 31.5, 31.5
Th1H (Heatsink Proximity) - 43.8125, 43.8125
Th2H () - 42.875, 42.875
Ts0P (Palm Rest) - 29.25, 29.25
Ts0S () - 37.4297, 37.4297
Ts1P () - 26.5625, 26.5625
Ts1S () - 34.9063, 34.9063
Ts2S () - 34.6016, 34.6016


Disk drives:
APPLE SSD SM0512L - 31

Path: /Applications/Macs Fan Control.app/Contents/MacOS/Macs Fan Control
**

**Comment by kleuter** [comment]
Can you please [record a screen video](https://support.apple.com/en-us/HT208721) beginning from the moment you launch the app and receive this message?

**Comment by Wickywow** [comment]
Sure


[Screenie.mp4.zip](https://github.com/crystalidea/macs-fan-control/files/3801874/Screenie.mp4.zip)

**Comment by kleuter** [comment]
Please try to locate error messages related to the app in the [Console](https://support.apple.com/en-us/guide/console/welcome/mac) after you receive this warning again. Thank you

**Comment by Wickywow** [comment]
Dont know if this means anything to you:

default	05:53:01.435411+0200	Macs Fan Control	SetFrontProcess: asn=0x0-0xc90c9 options=1
default	05:53:01.437489+0200	runningboardd	Acquiring assertion targeting executable<Macs Fan Control(503)> from originator [daemon<com.apple.coreservices.launchservicesd>:178] with description <RBSAssertionDescriptor; frontmost:3209; ID: 419-178-1652; target: 3209> attributes = {
    <RBSDomainAttribute: 0x7fee63b21fe0; domain: com.apple.launchservicesd; name: RoleUserInteractiveFocal; sourceEnvironment: 0x0>;
}
default	05:53:01.437732+0200	runningboardd	Assertion 419-178-1652 (target:executable<Macs Fan Control(503)>) will be created as active
default	05:53:01.439057+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring jetsam update because this process is not memory-managed
default	05:53:01.439450+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Set darwin role to: UserInteractiveFocal
default	05:53:01.439759+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring GPU update because this process is not GPU managed
default	05:53:01.441306+0200	runningboardd	Finished acquiring assertion 419-178-1652 (target:executable<Macs Fan Control(503)>)
default	05:53:07.919939+0200	runningboardd	Invalidating assertion 419-178-1652 (target:executable<Macs Fan Control(503)>) from originator 178
default	05:53:08.021192+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring jetsam update because this process is not memory-managed
default	05:53:08.021258+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Set darwin role to: UserInteractive
default	05:53:08.021301+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring GPU update because this process is not GPU managed
default	05:53:11.780857+0200	runningboardd	Acquiring assertion targeting executable<Macs Fan Control(503)> from originator [daemon<com.apple.coreservices.launchservicesd>:178] with description <RBSAssertionDescriptor; frontmost:3209; ID: 419-178-1654; target: 3209> attributes = {
    <RBSDomainAttribute: 0x7fee63a012e0; domain: com.apple.launchservicesd; name: RoleUserInteractiveFocal; sourceEnvironment: 0x0>;
}
default	05:53:11.781086+0200	runningboardd	Assertion 419-178-1654 (target:executable<Macs Fan Control(503)>) will be created as active
default	05:53:11.782635+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring jetsam update because this process is not memory-managed
default	05:53:11.784326+0200	runningboardd	Finished acquiring assertion 419-178-1654 (target:executable<Macs Fan Control(503)>)
default	05:53:11.783900+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Set darwin role to: UserInteractiveFocal
default	05:53:11.786081+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring GPU update because this process is not GPU managed
default	05:53:12.926817+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag **** frame data ****
default	05:53:12.926904+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames/s (IntervalRate)__ 0.0
default	05:53:12.926995+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames/s (IntervalRateIgnoringDelay)__ 0.0
default	05:53:12.927051+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms (OperationDuration)__ 1145.8
default	05:53:12.927092+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms/frame (AverageDuration)__ 0.0
default	05:53:12.927126+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms/frame (StandardDeviation)__ 0.0
default	05:53:12.927159+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms (DelayFromOperationStart)__ 0.0
default	05:53:12.927191+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames (IntervalCount)__ 0.0
default	05:53:19.223896+0200	runningboardd	Invalidating assertion 419-178-1654 (target:executable<Macs Fan Control(503)>) from originator 178
default	05:53:19.423074+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring jetsam update because this process is not memory-managed
default	05:53:19.423146+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Set darwin role to: UserInteractive
default	05:53:19.423223+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring GPU update because this process is not GPU managed
default	05:53:24.240452+0200	authd	Succeeded authorizing right 'system.privilege.admin' by client '/Applications/Macs Fan Control.app' [3209] for authorization created by '/Applications/Macs Fan Control.app' [3209] (13,0) (engine 254)
default	05:53:24.260302+0200	authd	Succeeded authorizing right 'com.apple.ServiceManagement.blesshelper' by client '/usr/libexec/smd' [286] for authorization created by '/Applications/Macs Fan Control.app' [3209] (3,0) (engine 255)
default	05:53:24.298892+0200	Macs Fan Control	SetFrontProcess: asn=0x0-0xc90c9 options=1
default	05:53:24.299691+0200	runningboardd	Acquiring assertion targeting executable<Macs Fan Control(503)> from originator [daemon<com.apple.coreservices.launchservicesd>:178] with description <RBSAssertionDescriptor; frontmost:3209; ID: 419-178-1659; target: 3209> attributes = {
    <RBSDomainAttribute: 0x7fee61e16560; domain: com.apple.launchservicesd; name: RoleUserInteractiveFocal; sourceEnvironment: 0x0>;
}
default	05:53:24.299858+0200	runningboardd	Assertion 419-178-1659 (target:executable<Macs Fan Control(503)>) will be created as active
default	05:53:24.301184+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring jetsam update because this process is not memory-managed
default	05:53:24.301548+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Set darwin role to: UserInteractiveFocal
default	05:53:24.302928+0200	runningboardd	Finished acquiring assertion 419-178-1659 (target:executable<Macs Fan Control(503)>)
default	05:53:24.302980+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring GPU update because this process is not GPU managed
default	05:53:33.305644+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag **** frame data ****
default	05:53:33.305699+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames/s (IntervalRate)__ 0.0
default	05:53:33.305788+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames/s (IntervalRateIgnoringDelay)__ 0.0
default	05:53:33.305826+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms (OperationDuration)__ 1715.8
default	05:53:33.305876+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms/frame (AverageDuration)__ 0.0
default	05:53:33.305915+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms/frame (StandardDeviation)__ 0.0
default	05:53:33.305944+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms (DelayFromOperationStart)__ 0.0
default	05:53:33.305979+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames (IntervalCount)__ 0.0
default	05:53:36.625920+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag **** frame data ****
default	05:53:36.625978+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames/s (IntervalRate)__ 0.0
default	05:53:36.626077+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames/s (IntervalRateIgnoringDelay)__ 0.0
default	05:53:36.626121+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms (OperationDuration)__ 374.4
default	05:53:36.626209+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms/frame (AverageDuration)__ 0.0
default	05:53:36.626257+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms/frame (StandardDeviation)__ 0.0
default	05:53:36.626295+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __ms (DelayFromOperationStart)__ 0.0
default	05:53:36.626333+0200	com.apple.PerformanceAnalysis.animationperfd	__PerfLogging__: Macs Fan Control__WSWindow__WSWindowDrag __frames (IntervalCount)__ 0.0
default	05:53:40.098320+0200	runningboardd	Invalidating assertion 419-178-1659 (target:executable<Macs Fan Control(503)>) from originator 178
default	05:53:40.203526+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring jetsam update because this process is not memory-managed
default	05:53:40.203598+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Set darwin role to: UserInteractive
default	05:53:40.203659+0200	runningboardd	[executable<Macs Fan Control(503)>:3209] Ignoring GPU update because this process is not GPU managed

**Comment by Wickywow** [comment]
Any idea yet?

**Comment by kleuter** [comment]
Looks like the [SMJobBless](https://developer.apple.com/documentation/servicemanagement/1431078-smjobbless?language=objc) call fails. Please try running the [latest beta](https://github.com/crystalidea/macs-fan-control/issues/218), it should output also the error code.
Must be fixed in the [latest beta](https://github.com/crystalidea/macs-fan-control/issues/218)

**Comment by Wickywow** [comment]
Yeah so this is the error with the new beta 1.5.1
<img width="416" alt="Screenshot 2019-11-05 at 19 17 21" src="https://user-images.githubusercontent.com/57282608/68230038-0067ee80-0001-11ea-8456-98dda9aaf0b0.png">

**Comment by kleuter** [comment]
Please run the following command in the Terminal and post a screenshot of the output
```
launchctl print-disabled system
```

**Comment by Wickywow** [comment]
<img width="520" alt="Screenshot 2019-11-06 at 06 39 27" src="https://user-images.githubusercontent.com/57282608/68268780-325c6d80-0060-11ea-959d-20fef8fabdca.png">
Here you go. I do use "little snitch" to monitor my outgoing connections but Macs Fan Control doesnt have any restrictions

**Comment by Wickywow** [comment]
<img width="956" alt="Screenshot 2019-11-06 at 06 48 56" src="https://user-images.githubusercontent.com/57282608/68269185-84ea5980-0061-11ea-9bbb-b355e9d2cf92.png">
This is my Little Snitch dash...its all allowed

**Comment by kleuter** [comment]
The error 9 is kSMErrorJobMustBeEnabled, which means the service is disabled.
launchctl output just verified that: the service 'com.crystalidea.macsfancontrol.smcwrite' is in the disabled list (for some UNKNOWN reason). Please try running the following command in the Terminal:
```
sudo launchctl enable system/com.crystalidea.macsfancontrol.smcwrite
```

It must help! Please accept my __big__ thanks for your research & help. The only open question is: **who did disable the service**?

**Comment by Wickywow** [comment]
DUDE!!!! You are a genius!! THANK YOU! It works now! No Idea who turned that off but Im just thankful its back on! Respect brother!

**Comment by kleuter** [comment]
Thanks a lot! I'll improve error handling for such scenario.

**Comment by kleuter** [comment]
Do you have any security software installed?

**Comment by kleuter** [comment]
fixed in 1.5.1

**Comment by VZcoding** [comment]
Same problem after update from catalina to big sur. Cannot connect to helper. "com.bjango.istatmenus.daemon" => false is always FALSE. "sudo launchctl enable system/com.crystalidea.macsfancontrol.smcwrite" does not turn it on. Tried to move app everywhere, nothing helps.. 

<img width="1036" alt="Screenshot 2020-11-24 at 18 15 42" src="https://user-images.githubusercontent.com/35932738/100121113-6d0a9100-2e81-11eb-992f-d0d563a66512.png">
